samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/ui/login/Login.kt

```diff
@@ -1,14 +1,184 @@
+import io.customer.android.sample.kotlin_compose.R
 
 @Composable
 fun LoginRoute(
     loginViewModel: LoginViewModel = hiltViewModel(),
     onSettingsClick: () -> Unit,
     onLoginSuccess: () -> Unit
 ) {
-    Text(text = "Login")
+    val state = loginViewModel.uiState.collectAsState()
+
+    LoginScreen(
+        uiState = state.value,
+        onLogin = { email, name ->
+            loginViewModel.loginUser(
+                email = email,
+                name = name,
+                onLoginSuccess = onLoginSuccess
+            )
+        },
+        onGuestLogin = {
+            loginViewModel.loginAsGuest(onLoginSuccess)
+        },
+        onSettingsClick = onSettingsClick
+    )
+}
+
+@Composable
+fun LoginScreen(
+    uiState: LoginUiState,
+    onLogin: (email: String, name: String) -> Unit,
+    onSettingsClick: () -> Unit,
+    onGuestLogin: () -> Unit
+) {
+    Box(modifier = Modifier.fillMaxSize()) {
+        Column(
+            verticalArrangement = Arrangement.spacedBy(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            LoginSettings(onSettingsClick)
+            LoginScreenHeader()
+            LoginFieldsView(
+                emailErrorState = if (uiState.emailError == null) "" else stringResource(id = uiState.emailError),
+                nameErrorState = if (uiState.nameError == null) "" else stringResource(id = uiState.nameError),
+                onLogin = onLogin,
+                onGuestLogin = onGuestLogin
+            )
+        }
+        VersionText()
+    }
+}
+
+@Composable
+fun ColumnScope.LoginSettings(onSettingsClick: () -> Unit) {
+    Icon(
+        imageVector = Icons.Default.Settings,
+        contentDescription = stringResource(R.string.settings),
+        modifier = Modifier
+            .size(72.dp)
+            .align(Alignment.End)
+            .padding(16.dp)
+            .clickable(onClick = onSettingsClick)
+    )
+}
+
+@Composable
+fun LoginScreenHeader() {
+    Text(
+        text = stringResource(R.string.ami_app),
+        style = MaterialTheme.typography.titleLarge
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun LoginFieldsView(
+    emailErrorState: String,
+    nameErrorState: String,
+    onLogin: (email: String, name: String) -> Unit,
+    onGuestLogin: () -> Unit
+) {
+    var name by remember { mutableStateOf("") }
+    var email by remember { mutableStateOf("") }
+
+    Column(
+        modifier = Modifier.padding(horizontal = 16.dp, vertical = 16.dp),
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        OutlinedTextField(
+            value = name,
+            onValueChange = { newValue -> name = newValue },
+            label = { Text(stringResource(R.string.name)) },
+            modifier = Modifier.fillMaxWidth().padding(start = 0.dp, end = 0.dp),
+            isError = nameErrorState.isNotEmpty(),
```

Comment:
I think QA should be able to test what happens to the SDK if an empty string is used as an identifier in the SDK. Not because our SDK should handle that scenario, but simply allow QA the ability to test this scenario. 

Therefore, I think that textfields in the app should not error if the textfield is empty.

Reply:
I don't particularly agree, it just unnecessarily breaks the user flow. If QA asks for this scenario we can create one for them. But we can't just guess and add it ourselves. 

I think others sample apps are doing the same except the FCM-COCOPODS one.